### PR TITLE
Convert test_conv_igemm_mlir_fwd, test_conv_igemm_mlir_bwd_wrw from CTest to gTest

### PR DESCRIPTION
**Pull request by xinlipn**
Convert two CTests to gTest, 
test_conv_igemm_mlir_fwd 
test_conv_igemm_mlir_bwd_wrw

### CODE REVIEW

**Comment by xinlipn** [comment]
@atamazov , could you help review when you have a min? Thanks

File: test/gtest/conv_igemm_mlir.cpp [hunk 1]
```diff
@@ -88,6 +88,7 @@ TEST_P(ConfigWithHalf, HalfTest)
     const auto& handle = get_handle();
     if((miopen::StartsWith(handle.GetDeviceName(), "gfx103") ||
         miopen::StartsWith(handle.GetDeviceName(), "gfx906")) &&
+       miopen::IsEnvvarValueEnabled("MIOPEN_TEST_MLIR") &&
```
Comment (atamazov) [inline test/gtest/conv_igemm_mlir.cpp:141]: [future] I think that we need to make this feature (is MLIR enabled or not) discoverable via some internal MIOpen API call, e.g. `miopen::debug::IsMlirEnabled()` and then use it in the tests instead of the env var. Because this env var is an additional dependence of gtests from ctest.

@JehandadKhan What do you think?

File: src/include/miopen/env.hpp [hunk 2]
```diff
@@ -50,20 +50,50 @@ inline bool IsEnvvarValueDisabled(const char* name)
 {
     // NOLINTNEXTLINE (concurrency-mt-unsafe)
     const auto value_env_p = std::getenv(name);
-    return value_env_p != nullptr &&
-           (std::strcmp(value_env_p, "disable") == 0 || std::strcmp(value_env_p, "disabled") == 0 ||
-            std::strcmp(value_env_p, "0") == 0 || std::strcmp(value_env_p, "no") == 0 ||
-            std::strcmp(value_env_p, "false") == 0);
+    if(value_env_p == nullptr)
```
Comment (atamazov) [inline src/include/miopen/env.hpp]: Please move this to a separate PR and update https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/docs/DebugAndLogging.md there as well.

File: src/include/miopen/env.hpp [hunk 3]
```diff
@@ -50,20 +50,50 @@ inline bool IsEnvvarValueDisabled(const char* name)
...
 
 inline bool IsEnvvarValueEnabled(const char* name)
 {
     // NOLINTNEXTLINE (concurrency-mt-unsafe)
     const auto value_env_p = std::getenv(name);
-    return value_env_p != nullptr &&
-           (std::strcmp(value_env_p, "enable") == 0 || std::strcmp(value_env_p, "enabled") == 0 ||
-            std::strcmp(value_env_p, "1") == 0 || std::strcmp(value_env_p, "yes") == 0 ||
-            std::strcmp(value_env_p, "true") == 0);
+    if(value_env_p == nullptr)
```
Comment (atamazov) [inline src/include/miopen/env.hpp]: Ditto

File: test/gtest/conv_igemm_mlir.cpp [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    if(precision == "--half")
+    {
+        test_cases.reserve(test_cases_bwd_wrw.size());
+        test_cases.insert(test_cases.end(), test_cases_bwd_wrw.begin(), test_cases_bwd_wrw.end());
+    }
+
+    return test_cases;
+}
+// Half for FWD, BWD, WRW
+INSTANTIATE_TEST_SUITE_P(ConvIgemmMlir, ConfigWithHalf, testing::Values(GetTestCases("--half")));
```
Comment (atamazov) [inline test/gtest/conv_igemm_mlir.cpp:244]: FLOAT_ENABLED is the default in the original test. Please add `--float` tests.

File: test/gtest/conv_2d.hpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "conv_common.hpp"
+
+template <class T>
+struct conv2d_driver : conv_driver<T>
+{
+    conv2d_driver() : conv_driver<T>()
```
Comment (atamazov) [inline test/gtest/conv_2d.hpp]: Please see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2168#discussion_r1223657449

File: test/gtest/conv_igemm_mlir.cpp [hunk 6]
```diff
@@ -0,0 +1,194 @@
+#include <tuple>
```
Comment (atamazov) [inline test/gtest/conv_igemm_mlir.cpp:26]: Add license header.

**Review by atamazov — CHANGES_REQUESTED** [review]

File: test/gtest/conv_2d.hpp (hunk 5 above)
Comment (xinlipn) [reply test/gtest/conv_2d.hpp]: > Please see [#2168 (comment)](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2168#discussion_r1223657449)

Updated

File: test/gtest/conv_igemm_mlir.cpp (hunk 6 above)
Comment (xinlipn) [reply test/gtest/conv_igemm_mlir.cpp:26]: > Add license header.

Updated

File: src/include/miopen/env.hpp (hunk 3 above)
Comment (xinlipn) [reply src/include/miopen/env.hpp]: > Ditto

This change has landed in develop already...

File: src/include/miopen/env.hpp (hunk 2 above)
Comment (xinlipn) [reply src/include/miopen/env.hpp]: > Please move this to a separate PR and update https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/docs/DebugAndLogging.md there as well.

The code changes have landed in develop already. PR created for document update
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2257

**Review by junliume — APPROVED** [review]
